src/olympia/discovery/views.py

```diff
@@ -22,6 +22,11 @@ def get_queryset(self):
         addons = self.queryset.in_bulk(ids)
 
         # Patch items to add addons.
+        result = []
         for item in discopane_items:
-            item.addon = addons[item.addon_id]
-        return discopane_items
+            try:
+                item.addon = addons[item.addon_id]
+                result.append(item)
+            except KeyError:
+                pass
```

Comment:
Since the list is hardcoded, would it hurt to log missing items? I'd imagine at least on prod we want to know if for some magic reason an item is missing in the list.